src/lib.rs

```diff
@@ -1,3 +1,4 @@
+#![feature(let_chains)]
```

Comment:
This should go, but I added it because it allowed me to confirm the tests I added pass with the current code.